src/helpers/parse-deep-link.ts

```diff
@@ -0,0 +1,103 @@
+import {utils} from 'ethers';
+
+export enum ValidUrlProtocol {
+  haqq = 'haqq',
+  etherium = 'etherium',
+  wc = 'wc',
+}
+
+export enum LinkType {
+  Address = 'address',
+  Etherium = 'etherium',
+  WalletConnect = 'wallet_connect',
+  Haqq = 'haqq',
+  Unrecognized = 'unrecognized',
+}
+
+type CommonResultData = {
+  rawData: string;
+};
+
+export type LinkParseResult = CommonResultData &
+  (
+    | {
+        type: LinkType.Address;
+        params: {
+          address: string;
+        };
+      }
+    | {
+        type: LinkType.Etherium;
+        params: {
+          address: string;
+        };
+      }
+    | {
+        type: LinkType.WalletConnect;
+        params: {
+          uri: string;
+        };
+      }
+    | {
+        type: LinkType.Haqq;
+        params: {
+          address?: string;
+          key?: string;
+          data?: any;
+        };
+      }
+    | {
+        type: LinkType.Unrecognized;
+        params: {};
+      }
+  );
+
+const isAddress = (link: string): boolean => {
+  return utils.isAddress(link);
+};
+
+export const parseDeepLink = (link: string): LinkParseResult => {
```

Comment:
Could you please add tests for this function with all cases? I think it's important to have safe deep link decoding because it may be used by attackers.